contracts/LightClientBridge.sol

```diff
@@ -0,0 +1,205 @@
+     * subsequent to the generation of this event before the 2nd tx can be sent
+     * @param prover The address of the calling prover
+     * @param blockNumber The blocknumber in which the initial validation
+     * succeeded
+     * @param id An identifier to provide disambiguation
+     */
+    event InitialVerificationSuccessful(address prover, uint256 blockNumber, uint256 id);
+
+    /**
+     * @notice Notifies an observer that the complete verification process has
+     *  finished successfuly and the block will be accepted
+     * @param prover The address of the successful prover
+     * @param statement the statement which was approved for inclusion
+     * @param id the identifier used
+     */
+    event FinalVerificationSuccessful(address prover, bytes32 statement, uint256 id);
+
+    /* Types */
+
+    struct ValidationData {
+        bytes32 statement;
+        uint256 validatorBitfield;
+        uint256 blockNumber;
+        uint256 id;
+    }
+
+    /* State */
+
+    address private validatorRegistry;
+    uint256 private count;
+    mapping(bytes32 => ValidationData) public validationData;
+
+    constructor () public {
+        validatorRegister = new ValidatorRegistry();
+        count = 0;
+    }
+
+    /* Public Functions */
+
+    /**
+     * @notice Called in order to register a validator
+     * @dev Calls out to a separate validatorRegistry contract to perform
+     * the logic
+     * @param validator A validator to register
+     */
+    function registerValidator(address validator) public onlyOwner {
+        validatorRegistry.registerValidator();
+    }
+
+    /**
+     * @notice Called in order to unregister a validator
+     * @dev Calls out to a separate validatorRegistry contract to perform
+     * the logic
+     * @param validator An array of validators to unregister
+     */
+    function unregisterValidator(address validator) public  {
+        validatorRegistry.unregisterValidator();
+    }
+
+    /**
+     * @notice Executed by the prover in order to begin the process of block
+     * acceptance by the light client
+     * @dev 
+     * @param statement contains the statement signed by the validator(s)
+     * @param validatorBitfield a bitfield containing the membership status of each
+     * validator who has claimed to have signed the statement
+     * @param sig the signature of an arbitrary validator
+     * @param proofs Proofs required for validation of the Merkle tree
+     */
+    function initialiseValidation(
+        bytes32 statement,
+        uint256 validatorBitfield,
+        bytes32 sig,
+        bytes32[] calldata proofs
+    ) public {
+        require(
+            validatorRegistry.checkValidatorInSet(),
+            "Error: Sender must be in validator set"
+        );
+
+        require(
+            validateSignature(statement, validatorBitfield, sig, proofs),
+            "Error: Invalid Signature"
+        );
+
+        count = SafeMath.add(count, 1);
```

Comment:
Add the statement `using SafeMath for uint256;` at the top of the contract and then update this to
```suggestion
        count = count.add(1);
```